c_src/scenic/comms.c

```diff
@@ -0,0 +1,480 @@
+  write_exact((byte*) key, msg_len);
+}
+
+//---------------------------------------------------------
+void send_dynamic_texture_miss(const char* key)
+{
+  uint32_t msg_len = strlen(key);
+  uint32_t cmd_len = msg_len + sizeof(uint32_t);
+  uint32_t cmd     = msg_out_dynamic_texture_miss;
+
+  ntoh_ui32(cmd_len);
+
+  write_exact((byte*) &cmd_len, sizeof(uint32_t));
+  write_exact((byte*) &cmd, sizeof(uint32_t));
+  write_exact((byte*) key, msg_len);
+}
+
+//---------------------------------------------------------
+void send_font_miss(const char* key)
+{
+  uint32_t msg_len = strlen(key);
+  uint32_t cmd_len = msg_len + sizeof(uint32_t);
+  uint32_t cmd     = msg_out_font_miss;
+
+  ntoh_ui32(cmd_len);
+
+  write_exact((byte*) &cmd_len, sizeof(uint32_t));
+  write_exact((byte*) &cmd, sizeof(uint32_t));
+  write_exact((byte*) key, msg_len);
+}
+
+//---------------------------------------------------------
+PACK(typedef struct msg_reshape_t
+{
+  uint32_t msg_id;
+  uint32_t window_width;
+  uint32_t window_height;
+}) msg_reshape_t;
+
+void send_reshape( int window_width, int window_height )
+{
+  msg_reshape_t msg = { msg_out_reshape, window_width, window_height };
+  write_cmd((byte*) &msg, sizeof(msg_reshape_t));
+}
+
+//---------------------------------------------------------
+PACK(typedef struct msg_key_t
+{
+  uint32_t msg_id;
+  uint32_t key;
+  uint32_t scancode;
+  uint32_t action;
+  uint32_t mods;
+}) msg_key_t;
+
+void send_key(int key, int scancode, int action, int mods)
+{
+  msg_key_t msg = { msg_out_key, key, scancode, action, mods };
+  write_cmd((byte*) &msg, sizeof(msg_key_t));
+}
+
+//---------------------------------------------------------
+PACK(typedef struct msg_codepoint_t
+{
+  uint32_t msg_id;
+  uint32_t codepoint;
+  uint32_t mods;
+}) msg_codepoint_t;
+
+void send_codepoint(unsigned int codepoint, int mods)
+{
+  msg_codepoint_t msg = { msg_out_codepoint, codepoint, mods };
+  write_cmd((byte*) &msg, sizeof(msg_codepoint_t));
+}
+
+//---------------------------------------------------------
+PACK(typedef struct msg_cursor_pos_t
+{
+  uint32_t msg_id;
+  float    x;
+  float    y;
+}) msg_cursor_pos_t;
+
+void send_cursor_pos(float xpos, float ypos)
+{
+  msg_cursor_pos_t msg = { msg_out_cursor_pos, xpos, ypos };
+  write_cmd((byte*) &msg, sizeof(msg_cursor_pos_t));
+}
+
+//---------------------------------------------------------
+PACK(typedef struct msg_mouse_button_t
+{
+  uint32_t msg_id;
+  uint32_t button;
+  uint32_t action;
+  uint32_t mods;
+  float    xpos;
+  float    ypos;
+}) msg_mouse_button_t;
+
+void send_mouse_button(int button, int action, int mods, float xpos, float ypos)
+{
+  msg_mouse_button_t msg = {
+    msg_out_mouse_button,
+    button,
+    action,
+    mods,
+    xpos,
+    ypos
+  };
+  write_cmd((byte*) &msg, sizeof(msg_mouse_button_t));
+}
+  // #define hton_ui16(x) (ntoh_ui16(x))
+  // #define hton_ui32(x) (ntoh_ui32(x))
+  // #define hton_f32(x) (ntoh_f32(x))
```

Comment:
Deadcode, possibly should be `#ifdef` code.

Reply:
Removed